### PR TITLE
chore: don't delete own data when leaving a project

### LINKED ISSUE #449: Do not delete own data by default when leaving a project
## Description

There are some lingering questions about what leaving a project entails when it comes to data deletion. For the sake of leaning towards the safer side for future work, we should update `MapeoProject[kLeaveProject]()` to use  `deleteOwn: false` when calling `CoreManager.deleteData()`.

## Tasks

- [ ] Update `MapeoProject[kLeaveProject]()` implementation

### PR DESCRIPTION
**Pull request by EvanHahn**
When you leave a project, we no longer delete your own data (but still
close your writer Hypercore).

Because this was the last occurrence of `deleteOwn: false`, I removed
the option entirely.

Fixes #449.

### CODE REVIEW

File: src/mapeo-project.js
Comment (achou11) [inline src/mapeo-project.js:623]: what's the context for this?

**Review by achou11 — APPROVED** [review]
have a non-blocking question but lgtm!

File: src/mapeo-project.js
Comment (achou11) [reply src/mapeo-project.js:623]: ah wait is it related to this note in your PR description?

> (but still close your writer Hypercore).

Comment (EvanHahn) [reply src/mapeo-project.js:623]: Yeah, that's exactly it. (Let me know if that's wrong!)

Comment (achou11) [reply src/mapeo-project.js:623]: i think that makes sense

Comment (EvanHahn) [reply src/mapeo-project.js:623]: I'm not 100% sure it's the right call. My goal is:

- When you leave, we want to close all Hypercores.
- When you leave, we want to delete all Hypercore data except yours.

Does this express that correctly? (And is my goal correct?)

**Comment by EvanHahn** [comment]
I'll plan to merge this tomorrow morning unless there are any objections.

**Pull request merged**
